### ISSUE TITLE
Using promises vs callbacks?

### ISSUE THREAD

**Issue by grofit** [jaggedsoft/node-binance-api#63]
Just noticed in 0.4.0 you are making the callbacks slightly different to account for error handling, and was wondering if there was an argument for just using promises instead of callbacks, as all modern browsers since IE and most node versions since around 0.11 (and easily pollyfillable if not available).

This would turn calls like:

```js
// Intervals: 1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w,1M
binance.candlesticks("BNBBTC", "5m", (error, ticks, symbol) => {
	if(error) { 
		console.error(error);
		return;
	}
	
	console.log("candlesticks()", ticks);
	let last_tick = ticks[ticks.length - 1];
	let [time, open, high, low, close, volume, closeTime, assetVolume, trades, buyBaseVolume, buyAssetVolume, ignored] = last_tick;
	console.log(symbol+" last close: "+close);
}, {limit: 500, endTime: 1514764800000});
```
To 

```js
// Intervals: 1m,3m,5m,15m,30m,1h,2h,4h,6h,8h,12h,1d,3d,1w,1M
binance.candlesticks("BNBBTC", "5m")
	.then(response => {
		const {ticks, symbol} = response;
		console.log("candlesticks()", ticks);
		const last_tick = ticks[ticks.length - 1];
		const [time, open, high, low, close, volume, closeTime, assetVolume, trades, buyBaseVolume, buyAssetVolume, ignored] = last_tick;
		console.log(symbol+" last close: "+close);
	})
	.catch(console.error);
```

**Comment by jaggedsoft** [comment]
Yes, I want to do this, just not sure how to make it work with `request`
It appears there is a way to make this work with the `request` library as shown here:
https://github.com/request/request-promise
https://github.com/request/request-promise/issues/132

Looking more in to this

https://github.com/binance-exchange/node-binance-api/issues/46

**Comment by jaggedsoft** [comment]
In the mean time there is another library that already supports this if necessary
https://github.com/binance-exchange/binance-api-node

**Comment by grofit** [comment]
No rush buddy, thanks for looking into it.

**Comment by elject** [comment]
Just a suggestion I know it might be too late to change libraries but...
You could use axios: https://github.com/axios/axios
It's an amazing library.
Parsing and promises already included.

**Comment by georgeMuraveiAlkh** [comment]
@grofit you could use manual promise wrappers like so: 
```
getOpenOrders () {
        return new Promise((resolve, reject) => {
            binance.openOrders(false, (error, openOrders) => {
                if(error !== null) return reject(error);
                resolve(openOrders);
            });
        })
    }

const checkPromise = async () => {
    console.log('wrapper works', await api.getOpenOrders())
}
checkPromise()
```
But it would be much more useful out of the box. Also without promises it is difficult to create good architecture which allows to slice API calls logic and trade strategy logic to use different APIs with the same trading strategy code (if, as example, I want to use my code on different exchange without any adjustments, only swapping api file)

**Comment by grofit** [comment]
Yeah, currently I am just using callbacks, but I use axios elsewhere so it would be handy for using `await` as well as being able to chain things and handle exceptions slightly nicer.

**Comment by rafaelklaessen** [comment]
It's also possible to wrap `request` calls with [promisify](https://www.npmjs.com/package/es6-promisify)

**Comment by rafaelklaessen** [comment]
Using promises would fix #74 I suppose
